ruma-push-gateway-api/src/push/send_event_notification/v1.rs

```diff
@@ -118,64 +118,25 @@ pub struct Notification<'a> {
     /// This is a dictionary of the current number of unacknowledged
     /// communications for the recipient user. Counts whose value is zero should
     /// be omitted.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub counts: Option<NotificationCounts>,
+    #[serde(skip_serializing_if = "ruma_serde::is_default")]
+    pub counts: NotificationCounts,
 
     /// This is an array of devices that the notification should be sent to.
     pub devices: &'a [Device],
 }
 
 impl<'a> Notification<'a> {
-    /// Create a new notification with:
-    /// * the given event id.
-    /// * the given room id.
-    /// * the given event_type.
-    /// * the notification sender
-    /// * the notification sender's display name
-    /// * the room's name
-    /// * the room's alias
-    /// * whether the user is the target of the notification
-    /// * the priority of the notification
-    /// * the content of the notification
-    /// * the number of unacknowledged communications for the recipient
-    /// * the devices to send the notification to
-    #[allow(clippy::too_many_arguments)]
-    pub fn new(
-        event_id: Option<&'a EventId>,
-        room_id: Option<&'a RoomId>,
-        event_type: Option<&'a EventType>,
-        sender: Option<&'a UserId>,
-        sender_display_name: Option<&'a str>,
-        room_name: Option<&'a str>,
-        room_alias: Option<&'a RoomAliasId>,
-        user_is_target: Option<bool>,
-        prio: Option<NotificationPriority>,
-        content: Option<Box<RawJsonValue>>,
-        counts: Option<NotificationCounts>,
-        devices: &'a [Device],
-    ) -> Self {
-        Notification {
-            event_id,
-            room_id,
-            event_type,
-            sender,
-            sender_display_name,
-            room_name,
-            room_alias,
-            user_is_target,
-            prio,
-            content,
-            counts,
-            devices,
-        }
+    /// Create a new notification for the given devices
+    pub fn new(devices: &'a [Device]) -> Self {
+        Notification { devices, ..Default::default() }
     }
 }
 
 /// Type for passing information about notification priority.
 ///
 /// This may be used by push gateways to deliver less time-sensitive
 /// notifications in a way that will preserve battery power on mobile devices.
-#[derive(Clone, Copy, Debug, Serialize, Deserialize, Display, EnumString)]
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, Display, EnumString, PartialEq)]
```

Comment:
We don't generally impl any of `Ord, Eq` traits, I think this is to keep code-gen/compilation time to a down.

Reply:
In this case `PartialEq` was added for `ruma_serde::is_default`